collector/lib/CollectionMethod.h

```diff
@@ -0,0 +1,35 @@
+/** collector
+
+A full notice with attributions is provided along with this source code.
+
+This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License version 2 as published by the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+* In addition, as a special exception, the copyright holders give
+* permission to link the code of portions of this program with the
+* OpenSSL library under certain conditions as described in each
+* individual source file, and distribute linked combinations
+* including the two.
+* You must obey the GNU General Public License in all respects
+* for all of the code used other than OpenSSL.  If you modify
+* file(s) with this exception, you may extend this exception to your
+* version of the file(s), but you are not obligated to do so.  If you
+* do not wish to do so, delete this exception statement from your
+* version.
+*/
+
+#ifndef COLLECTION_METHOD_H
+#define COLLECTION_METHOD_H
+
+namespace collector {
+enum collectionMethod {
```

Comment:
[nit] should probably be TitleCase

Reply:
I kinda have a collision problem, since some classes have a `CollectionMethod` method that acts as a getter, there are 2 possible solutions:
- We prefix the type were needed as `enum CollectionMethod`
- We rename the getters to be `GetCollectionMethod`

I think the second option is nicer and should make it easier when grepping for it.

Reply:
I agree, I think the second option is best in this case